tests/TaskManagerTest.php

```diff
@@ -563,6 +564,21 @@ public function testTagNameNotAllowed()
         $this->taskManager->save($task1);
     }
 
+    public function testProjectHierarchyAllowed()
+    {
+        $projects = ['grandparent.parent.child', 'parent.child'];
+        foreach ($projects as $project) {
+            $task = new Task();
+            $task->setDescription('foo1');
+            $task->setProject($project);
+
+            // now exception thrown
+            $this->taskManager->save($task);
```

Comment:
you test only the write operation and not read. you should try fetch a new instance and check the project.

```php
$this->taskManager->clear();
$task = $this->taskManager->find($task->getUuid);
// ...
```